Source/ACE.Server/WorldObjects/Vendor.cs

```diff
@@ -72,12 +73,12 @@ public override void ActOnUse(WorldObject wo)
 
             var rotateTime = Rotate(player);    // vendor rotates towards player
 
-            // TODO: remove this when DelayManager is not forward propagating current tick time
+            var delayUntilTime = Time.GetFutureUnixTime(rotateTime);
 
             var actionChain = new ActionChain();
-            actionChain.AddDelaySeconds(0.001f);  // force to run after rotate.EnqueueBroadcastAction
```

Comment:
This doesn't fix the core problems with ActionChain

var firstChain = new ActionChian();
firstChain.AddAction(this, SomeAction);
firstChain.EnqueueChain();

var secondChain = new ActionChain();
secondChain.AddAction(this, SomeOtherAction);
secondChain.EnqueueChain();

Why is SomeOtherAction running before SomeAction in that case? The actions should be processed in the order they are added

AddDelayForOneTick is not something that should be needed anywhere

Please see https://github.com/gmriggs/ACE/blob/60fa9c0d9fafba775f127183be18dc556f174d70/Source/ACE.Server/Entity/Actions/ActionChain.cs for how ActionChains should work

Reply:
That is incorrect.

Action chains enqueued on the same IActor are run in the order in which they are enqueued.